app/controllers/api/v2/courses_controller.rb

```diff
@@ -168,14 +168,19 @@ def update_sites
       def update_subjects
         return if subject_ids.nil?
 
-        @course.subjects = []
-        @course.subjects = subject_ids.map { |id| Subject.find(id) }
+        if request_has_duplicate_subject_ids?
+          @course.errors.add(:subjects, :duplicate)
+        else
+          @course.subjects = []
 
-        subject_ids.each_with_index do |id, index|
-          @course.course_subjects.select { |cs| cs.subject_id == id.to_i }.first.position = index
-        end
+          @course.subjects = subject_ids.map { |id| Subject.find(id) }
 
-        @course.name = @course.generate_name
+          subject_ids.each_with_index do |id, index|
+            @course.course_subjects.select { |cs| cs.subject_id == id.to_i }.first.position = index
+          end
```

Comment:
Feels like this belongs in `CourseSubjects` or a service object

Reply:
I did consider a service/on the model but this seems like behaviour specific to the API layer rather than anywhere else :thinking: cos it's handling IDs given in over the API